### PR TITLE
Sensor history

### PR DESCRIPTION
**Pull request by kutluo**
This should work, but I did not have the opportunity to test it with real data. Haven't tested the mongo service method but the response should be fine.

### CODE REVIEW

File: pems-server/src/main/java/org/maroubra/pemsserver/api/models/sensors/responses/SensorHistoryResponse.java [hunk 1]
```diff
@@ -1,4 +1,37 @@
 package org.maroubra.pemsserver.api.models.sensors.responses;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import org.maroubra.pemsserver.monitoring.SensorLog;
+
+import java.util.ArrayList;
+import java.util.List;
+
 public class SensorHistoryResponse {
```
Comment (twgraham) [inline pems-server/src/main/java/org/maroubra/pemsserver/api/models/sensors/responses/SensorHistoryResponse.java]: I'm not sure the resulting JSON is what we would want? Can you give an example JSON you'd expect from this? Are we matching type, value and time by index?

File: pems-server/src/main/java/org/maroubra/pemsserver/api/resources/SensorsResource.java [hunk 2]
```diff
@@ -77,12 +79,13 @@ public void updateSensor(@PathParam("id") String id, @ApiParam(name = "JSON body
...
-    @Path("{id}/history")
+    @Path("{id}/{dataSize}/history")
     @ApiOperation(value = "Get a sensors history")
     @ApiResponses(value = {
-            @ApiResponse(code = 404, message = "Specified actuator does not exist")
+            @ApiResponse(code = 404, message = "Specified sensor does not exist")
     })
-    public SensorHistoryResponse getHistory(@PathParam("id") String id) {
-        throw new UnsupportedOperationException();
+    public SensorHistoryResponse getHistory(@PathParam("id") String id, @PathParam("dataSize") int size) {
```
Comment (twgraham) [inline pems-server/src/main/java/org/maroubra/pemsserver/api/resources/SensorsResource.java]: Chang dataSize to a query param, and provide defaults. maybe 50?

Comment (twgraham) [reply pems-server/src/main/java/org/maroubra/pemsserver/api/resources/SensorsResource.java]: So the `@ApiPath` would look like "{id}/history", and you can just label the `dataSize` as a query parameter instead

File: pems-server/src/main/java/org/maroubra/pemsserver/api/models/sensors/responses/SensorHistoryResponse.java (hunk 1 above)
Comment (kutluo) [reply pems-server/src/main/java/org/maroubra/pemsserver/api/models/sensors/responses/SensorHistoryResponse.java]: currently, the JSON  will look like this, the reason I did that was to accommodate for values from the accelerometer.
{
   "valueName":[
      [
         "BAT"
      ],
      [
         "accel-x",
         "accel-y",
         "accel-z"
      ]
   ],
   "value":[
      [
         91
      ],
      [
         0.9,
         5,
         0.3
      ]
   ],
   "timeStamp":[
      "2017-10-28T10:06:11.859",
      "2017-10-29T10:06:11.859"
   ]
}
However we can make it look like this by changing add to addAll.
{
   "valueName":[
      "BAT",
      "accel-x",
      "accel-y",
      "accel-z"
   ],
   "value":[
      91,
      0.9,
      5,
      0.3
   ],
   "timeStamp":[
      "2017-10-28T10:09:08.318",
      "2017-10-29T10:09:08.321"
   ]
}

Comment (kutluo) [reply pems-server/src/main/java/org/maroubra/pemsserver/api/models/sensors/responses/SensorHistoryResponse.java]: or I can put in checks and combine both of them together.

File: pems-server/src/main/java/org/maroubra/pemsserver/monitoring/MonitoringServiceImpl.java [hunk 3]
```diff
@@ -69,6 +70,10 @@ public Completable createSensor(SensorConfig config) {
         return sensorConfigCollection.insertOne(config).toCompletable().doOnCompleted(() -> startSensor(sensor));
     }
 
+    public List<SensorLog> getSensorLogs(String sensorId, int limit) {
+        return sensorLogsCollection.find(Filters.eq("sensorID",sensorId)).limit(limit).toObservable().toList().toBlocking().single();
```
Comment (twgraham) [inline pems-server/src/main/java/org/maroubra/pemsserver/monitoring/MonitoringServiceImpl.java]: I think it should be "sensorId" not "sensorID" (naming in the config class)

Comment (kutluo) [reply pems-server/src/main/java/org/maroubra/pemsserver/monitoring/MonitoringServiceImpl.java]: fixed

**Pull request merged**